client/src/reducers/autosave.ts

```diff
@@ -0,0 +1,92 @@
+import type { AnyAction } from "redux";
+
+import type AnnoMatrix from "../annoMatrix/annoMatrix";
+import type { RootState } from ".";
+import type { Genesets } from "./genesets";
+
+export interface AutosaveState {
+  obsAnnotationSaveInProgress: boolean;
+  lastSavedAnnoMatrix: AnnoMatrix | null;
+  genesetSaveInProgress: boolean;
+  lastSavedGenesets: Genesets | null;
+  error: string | false;
+}
+
+const initialState: AutosaveState = {
```

Comment:
Autosave reducer: please check initialization and that save completion always advances lastSaved* to prevent churn; error handling semantics look reasonable?